utils/filesystem/files.go

```diff
@@ -1129,6 +1128,60 @@ func (fs *VFS) LsWithExclusionPatterns(dir string, exclusionPatterns ...string)
 	return
 }
 
+func (fs *VFS) LsRecursive(ctx context.Context, dir string) (files []string, err error) {
+	limits := NoLimits()
+	return fs.LsRecursiveWithExtensionPatternsAndLimits(ctx, dir, limits)
+}
+
+func (fs *VFS) LsRecursiveWithExtensionPatterns(ctx context.Context, dir string, exclusionPatterns ...string) (files []string, err error) {
+	limits := NoLimits()
+	return fs.LsRecursiveWithExtensionPatternsAndLimits(ctx, dir, limits, exclusionPatterns...)
+}
+
+func (fs *VFS) LsRecursiveWithExtensionPatternsAndLimits(ctx context.Context, dir string, limits ILimits, exclusionPatterns ...string) (files []string, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	if limits == nil {
+		err = fmt.Errorf("%w: missing file system limits definition", commonerrors.ErrUndefined)
```

Comment:
it isn't a definition
```suggestion
		err = fmt.Errorf("%w: missing file system limits", commonerrors.ErrUndefined)
```